.github/workflows/build.yml

```diff
@@ -4,10 +4,7 @@ name: Nextflow CI
 on:
   push:
     branches:
-      - '*'
-      - '!refs/tags/.*'
-    tags-ignore:
-      - '*'
+      - 'master'
```

Comment:
umm. but this will disable tests on branches other than master

Reply:
I think it disables tests on pushes to other branches, but pull requests will still be tested, see the `pull_request` below 👇

Reply:
But it's a wrong assumption! it's also useful to push a branch and have tests run even without a PR

Reply:
Just make a PR

Reply:
or, what if we add a manual trigger? that way you can test branches only when you want to. also could be a workaround for when PR tests mysteriously don't run

Reply:
Added 👍🏻

Reply:
If the change doesn't match your workflow (you often merge branches manually / you like seeing CI on branch pushes during development) then we can change or just close. I opened the PR as a draft initially just to propose the suggestion.

Reply:
Would the CI be trigged when pushing new commit to a PR (both internal or external) ?

Reply:
If the PR is already open, then yes in both cases 👍🏻

Reply:
```suggestion
      - 'master'
      - 'testing*'
      - 'dev*'
```

Ok, are patterns in the branch names supported? If so it could be a nice compromise

Reply:
Let's try it!

Reply:
@pditommaso you didn't commit the suggestion before merging! 😅

Reply:
<img src="https://media1.giphy.com/media/p60L4kNY1szqU/giphy.gif"/>

Reply:
Added here: https://github.com/nextflow-io/nextflow/pull/3852